src/stylefunction.js

```diff
@@ -554,7 +556,28 @@ export function stylefunction(
     cameraObj.zoom = zoom;
     cameraObj.distanceFromCenter = 0;
     const featureGeometry = feature.getGeometry();
+
+    if (featureGeometry instanceof GeometryCollection) {
+      const geoms = featureGeometry.getGeometries();
+
+      for (const geom of geoms) {
+        const subFeature = new Feature({
+          ...feature.getProperties(),
+          geometry: geom,
+        });
+
+        const subStyles = styleFunction(subFeature, resolution, onlyLayer);
+
+        if (subStyles && Array.isArray(subStyles)) {
```

Comment:
No `Array.isArray()` check necessary here. `styleFunction()` returns `undefined` or an array of styles.